statistics/handle/handle.go

```diff
@@ -383,9 +391,13 @@ func (h *Handle) MergePartitionStats2GlobalStats(sc *stmtctx.StatementContext, i
 		// Merge topN. We need to merge TopN before merging the histogram.
 		// Because after merging TopN, some numbers will be left.
 		// These left numbers should be inserted into the histogram.
-		err = errors.Errorf("TODO: The merge function of the topN structure has not been implemented yet")
-		if err != nil {
-			return
+		var popedTopN []statistics.TopNMeta
+		globalStats.TopN[i], popedTopN = statistics.MergeTopN(allTopN[i], 0)
+
+		if len(popedTopN) != 0 {
+			popedTopNHg := statistics.NewHistogram(allID[i], 0, 0, 0, allFieldType[i], 0, 0)
+			popedTopNHg.AddIdxVals(popedTopN)
```

Comment:
There seems to be a problem with the `AddIdxVals`. If the value is greater or less than the boundaries of all buckets, it cannot be inserted.

Reply:
Update. PTAL